src/main/java/com/github/kagkarlsson/scheduler/SchedulerClient.java

```diff
@@ -27,9 +28,13 @@
 public interface SchedulerClient {
 	
 	void scheduleForExecution(Instant exeecutionTime, TaskInstance taskInstance);
```

Comment:
Maybe we should do some renaming here. 
How about:
* schedule
* reschedule
* cancel

And maybe a default method for the old scheduleForExecution, possibly deprecated?